### ISSUE TITLE
Mismatched version numbers between table and descriptions in "Compiler Warnings by compiler version"

### ISSUE THREAD

**Issue by danieljennings** [MicrosoftDocs/cpp-docs#3563]
https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warnings-by-compiler-version?view=msvc-170

This page has two components, a table and a per-version description, and they don't agree about verseion numbers for certain VS2019 builds.

Specifically, the table has these "Compiler version number" specified:
```
Visual Studio 2019 version 16.7  | 19.27.29112.0
Visual Studio 2019 version 16.8  | 19.28.29330.0
Visual Studio 2019 version 16.9  | 19.28.29500.0
Visual Studio 2019 version 16.10 | 19.28.30000.0
Visual Studio 2019 version 16.11 | 19.28.30100.0
```

Which disagrees with the compiler versions listed by the subsequent headers:
```
Warnings introduced in Visual Studio 2019 version 16.7  (compiler version 19.27.29112.0)
Warnings introduced in Visual Studio 2019 version 16.8  (compiler version 19.28.29330.0)
Warnings introduced in Visual Studio 2019 version 16.9  (compiler version 19.28.29700.0)
Warnings introduced in Visual Studio 2019 version 16.10 (compiler version 19.29.30000.0)
Warnings introduced in Visual Studio 2019 version 16.11 (compiler version 19.29.30100.0)
```

Specifically, the second component of the version mismatches for 16.10 and 16.11 (28 vs 29), while the third component mismatches for 16.9 (29500 vs 29700).

Furthermore, depending on whether the table is wrong or the header, 16.9's section says to disable those warnings by specifying `/Wv:19.28.29500` which is the same version it lists in the table. If 16.9 is actually 19.28.29500 then I imagine that section is supposed to say you need to specify `/Wv:19.28.29330.0` to anchor to 16.8's warnings.

Finally, there's nothing on this page for VS2022 - were there no new compiler warnings added to that version?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 672bc066-51a0-9778-4c50-98922ebb4b4c
* Version Independent ID: 6263d065-c81a-54f7-a99d-fef41134c872
* Content: [Compiler Warnings by compiler version](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warnings-by-compiler-version?view=msvc-170)
* Content Source: [docs/error-messages/compiler-warnings/compiler-warnings-by-compiler-version.md](https://github.com/Microsoft/cpp-docs/blob/master/docs/error-messages/compiler-warnings/compiler-warnings-by-compiler-version.md)
* Product: **visual-cpp**
* Technology: **cpp-diagnostics**
* GitHub Login: @corob-msft
* Microsoft Alias: **corob**

**Comment by corob-msft** [comment]
@danieljennings 
Thanks for pointing out the discrepancies. The last two entries in the table should be numbered starting with 19.29. I've fixed them in an update that should appear in a day or so. We try to update the errors and warnings docs on the same cycle as the product, but sometimes we're a little late. These updates will include Visual Studio 2022 version 17.0 and 17.1 diagnostics entries.

The statements about the **`/Wv`** option to use in each section sometimes account for later build numbers of a particular version than the initial build number listed in the table at the top. Since **`/Wv:version.number`** means "show me errors from versions that start with *version.number* or less" and some versions share major and minor numbers, we provide a number that should include all servicing updates for the previous version. That number was complicated by the 16.9, 16.10, and 16.11 releases of Visual Studio, which made those distinctions down in the less significant bits of the build numbers. I've checked these numbers and made adjustments to a couple of them to track updates correctly. Thanks for the input.
